File: aspnet-in-practice/08-09/VB/CoolMVCBlog/Models/CommentMetadata.vb
Imports System.ComponentModel.DataAnnotations

Public Class CommentMetadata

    <Required(ErrorMessage:="Author is mandatory")>
    Public Property Author As String

    <Required(ErrorMessage:="Email is mandatory")>
    <RegularExpression("\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}\b", ErrorMessage:="Please provide a valid Email address")>
    Public Property Email As String

    <Required(ErrorMessage:="You should post some text")>
    <DataType(DataType.MultilineText)>
    Public Property Text As String
End Class

<MetadataType(GetType(CommentMetadata))>
Partial Public Class Comment

End Class

File: aspnet-in-practice/08-09/CS/CoolMVBlog/Models/CommentMetadata.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;
using CoolMvcBlog.Utils;

namespace CoolMvcBlog.Models
{
    public class CommentMetadata
    {
        [Required(ErrorMessage="Author is mandatory")]
        public string Author { get; set; }

        [Required(ErrorMessage="Email is mandatory")]
        [EmailAddress]
        [RegularExpression(@"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}\b", ErrorMessage="Please provide a valid Email address")]
        public string Email { get; set; }

        [Required(ErrorMessage="You should post some text")]
        [DataType(DataType.MultilineText)]
        public string Text { get; set; }
    }

    [MetadataType(typeof(CommentMetadata))]
    public partial class Comment
    { }
}